pkg/pillar/types/global.go

```diff
@@ -114,6 +117,166 @@ func (gs *GlobalStatus) UpdateItemValuesFromGlobalConfig(gc GlobalConfig) {
 // Agents subscribe to this info to get at least the log levels
 // A value of zero means we should use the default
 // All times are in seconds.
+type ConfigItemType uint8
+
+const (
+	CONFIGITEM_TYPE_INT ConfigItemType = iota + 1
+	CONFIGITEM_TYPE_BOOL
+	CONFIGITEM_TYPE_STRING
+	CONFIGITEM_TYPE_TRISTATE
+	CONFIGITEM_TYPE_AGENTSETTING
```

Comment:
Hmm.. This doesn't belong here.. each AGENT_SETTING itself can be any of the 4 types above..